pkg/apis/application/v1alpha1/types.go

```diff
@@ -671,6 +671,11 @@ type HelmRepository struct {
 	Password string `json:"password,omitempty" protobuf:"bytes,7,opt,name=password"`
 }
 
+type ResourceOverride struct {
```

Comment:
It feels weird to have this in types.go, how come it was moved here?

Reply:
It is here since I had to expose it via API. Otherwise, we would have to maintain one more generated.proto file.

Reply:
But the settings.proto is where I expected this to be exposed, similar to OIDCConfig, DexConfig.  Can it be moved there?

Reply:
This would create a recursive dependency between `server/settings` and `util/settings` packages.

Unless I'm missing something we would have the same problem with any data structure which is used in  `business logic` component and exposed via API.